app/models/manageiq/providers/ansible_tower/automation_manager/credential_mixin.rb

```diff
@@ -0,0 +1,134 @@
+      :required   => false,
+      :label      => _('Security Token'),
+      :help_text  => _('Security Token for this credential'),
+      :max_length => 1024,
+      :encrypted  => true,
+      :default    => ''
+    },
+    :ssh_key_data     => {
+      :type       => 'string',
+      :required   => false,
+      :label      => _('SSH private key'),
+      :help_text  => _('RSA or DSA private key to be used instead of password.'),
+      :encrypted  => true,
+      :default    => ''
+    },
+    :ssh_key_unlock   => {
+      :type       => 'string',
+      :required   => false,
+      :label      => _('SSH key unlock'),
+      :help_text  => _('Passphrase to unlock SSH private key if encrypted (or \'ASK\' to prompt the user for machine credentials).'),
+      :max_length => 1024,
+      :encrypted  => true,
+      :default    => ''
+    },
+    :become_method    => {
+      :type       => 'choice',
+      :required   => false,
+      :label      => _('Become method'),
+      :help_text  => _('Privilege escalation method.'),
+      :encrypted  => false,
+      :default    => '',
+      :choices    => %w(sudo su pbrun pfexec).insert(0, '')
+    },
+    :become_username  => {
+      :type       => 'string',
+      :required   => false,
+      :label      => _('Become username'),
+      :help_text  => _('Privilege escalation username.'),
+      :max_length => 1024,
+      :encrypted  => false,
+      :default    => ''
+    },
+    :become_password => {
+      :type       => 'string',
+      :required   => false,
+      :label      => _('Become password'),
+      :help_text  => _('Password for privilege escalation method.'),
+      :max_length => 1024,
+      :encrypted  => true,
+      :default    => ''
+    },
+    :vault_password => {
+      :type       => 'string',
+      :required   => false,
+      :label      => _('Vault password'),
+      :help_text  => _('Vault password (or \'ASK\' to prompt the user).'),
+      :max_length => 1024,
+      :encrypted  => true,
+      :default    => ''
+    },
+    :host => {
+      :type       => 'string',
+      :required   => false,
+      :label      => _('Host'),
+      :help_text  => _('The hostname or IP address to use.'),
+      :max_length => 1024,
+      :encrypted  => false,
+      :default    => ''
+    }
+  }.freeze
+
+  included do
+    self::ATTRIBUTES.each do |attr|
+      define_method(:"#{attr}") do
+        options && options[attr] ? options[attr] : FIELDS[attr][:default]
+      end
+
+      define_method(:"#{attr}=") do |value|
+        self.options ||= {}
+        self.options[attr] = value
+      end
+
+      define_method(:"#{attr}_meta") do
+        FIELDS[attr]
+      end
+
+      [:type, :label, :help_text, :max_length, :encrypted, :choices].each do |meta|
+        define_method(:"#{attr}_#{meta}") do
+          FIELDS[attr][meta]
+        end
+      end
+    end
+  end
+
+  module ClassMethods
+    def fields
+      self::ATTRIBUTES.each_with_object({}) do |field_key, hash|
```

Comment:
There is no need to rebuild this on every call.  It should probably be in a constant.